File: src/main/java/com/github/sorusclient/client/module/impl/blockoverlay/v1_8_9/BlockOverlayTransformer.java
package com.github.sorusclient.client.module.impl.blockoverlay.v1_8_9;

import com.github.glassmc.loader.GlassLoader;
import com.github.glassmc.loader.Listener;
import com.github.glassmc.loader.util.Identifier;
import com.github.sorusclient.client.transform.Applier;
import com.github.sorusclient.client.transform.Transformer;
import org.objectweb.asm.Opcodes;
import org.objectweb.asm.tree.*;

public class BlockOverlayTransformer extends Transformer implements Listener {

    @Override
    public void run() {
        GlassLoader.getInstance().registerTransformer(BlockOverlayTransformer.class);
    }

    public BlockOverlayTransformer() {
        this.setHookClass(BlockOverlayHook.class);
        this.register("v1_8_9/net/minecraft/client/render/WorldRenderer", this::transformWorldRenderer);
    }

    private void transformWorldRenderer(ClassNode classNode) {
        Identifier method_9895 = Identifier.parse("v1_8_9/net/minecraft/client/render/WorldRenderer#method_9895(Lv1_8_9/net/minecraft/util/math/Box;)V");

        this.findMethod(classNode, method_9895)
                .apply(new Applier.Insert<>(this.createList(insnList -> {
                    insnList.add(new VarInsnNode(Opcodes.ALOAD, 0));
                    insnList.add(this.getHook("preRenderOutline"));
                })));
    }

}

File: src/main/kotlin/com/github/sorusclient/client/feature/impl/blockoverlay/v1_8_9/BlockOverlayTransformer.kt
package com.github.sorusclient.client.feature.impl.blockoverlay.v1_8_9

import com.github.glassmc.loader.api.GlassLoader
import com.github.glassmc.loader.api.Listener
import com.github.glassmc.loader.util.Identifier
import com.github.sorusclient.client.transform.Applier
import com.github.sorusclient.client.transform.Transformer
import org.objectweb.asm.Opcodes
import org.objectweb.asm.tree.*

class BlockOverlayTransformer: Transformer(), Listener {

    override fun run() {
        GlassLoader.getInstance().registerTransformer(BlockOverlayTransformer::class.java)
    }

    init {
        setHookClass(BlockOverlayHook::class.java)
        register("v1_8_9/net/minecraft/client/render/WorldRenderer", this::transformWorldRenderer)
    }

    private fun transformWorldRenderer(classNode: ClassNode) {
        val method9895 = Identifier.parse("v1_8_9/net/minecraft/client/render/WorldRenderer#method_9895(Lv1_8_9/net/minecraft/util/math/Box;)V")
        findMethod(classNode, method9895)
            .apply(Applier.Insert(createList { insnList ->
                insnList.add(VarInsnNode(Opcodes.ALOAD, 0))
                insnList.add(this.getHook("onBlockOverlayRender"))
            }))
    }

}